lib/jsdom/living/window/navigation.js

```diff
@@ -67,37 +67,103 @@ exports.traverseHistory = (window, specifiedEntry, flags) => {
   window._currentSessionHistoryEntryIndex = window._sessionHistory.indexOf(specifiedEntry);
 };
 
-exports.navigate = (window, newURL) => {
+// https://html.spec.whatwg.org/#navigating-across-documents
+exports.navigate = (window, newURL, flags) => {
   // This is NOT a spec-compliant implementation of navigation in any way. It implements a few selective steps that
   // are nice for jsdom users, regarding hash changes. We could maybe implement javascript: URLs in the future, but
   // the rest is too hard.
 
   const document = idlUtils.implForWrapper(window._document);
-
   const currentURL = document._URL;
 
-  if (newURL.scheme !== currentURL.scheme ||
-      newURL.username !== currentURL.username ||
-      newURL.password !== currentURL.password ||
-      newURL.host !== currentURL.host ||
-      newURL.port !== currentURL.port ||
-      !arrayEqual(newURL.path, currentURL.path) ||
-      newURL.query !== currentURL.query ||
-      // Omitted per spec: url.fragment !== this._url.fragment ||
-      newURL.cannotBeABaseURL !== currentURL.cannotBeABaseURL) {
-    notImplemented("navigation (except hash changes)", window);
+  if (!flags.reloadTriggered && urlEquals(currentURL, newURL, { excludeFragments: true })) {
+    if (newURL.fragment !== currentURL.fragment) {
+      navigateToFragment(window, newURL, flags);
+    }
     return;
   }
 
-  if (newURL.fragment !== currentURL.fragment) {
-    // https://html.spec.whatwg.org/#scroll-to-fragid
+  // NOT IMPLEMENTED: Prompt to unload the active document of browsingContext.
 
-    window._sessionHistory.splice(window._currentSessionHistoryEntryIndex + 1, Infinity);
+  window.close();
```

Comment:
Doing this unconditionally seems like a huge behavior change. I don't think we should.